frontend/src/hooks/useDeepCompareCallback.ts

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { useDeepCompareMemoize } from './useDeepCompareMemoize';
+
+/**
+ * `useDeepCompareCallback` will return a memoized version of the callback that
+ * only changes if one of the `deps` has changed.
+ *
+ * Usage note: only use this if `deps` are objects or arrays that contain
+ * objects. Otherwise you should just use React.useEffect.
```

Comment:
Wouldn't this make more sense instead of `React.useEffect` it would be `React.useCallback`?